src/lib/components/ui/input/date-input.svelte

```diff
@@ -1,8 +1,352 @@
+	let selectedYear = $state(currentDate.getFullYear())
+	let selectedDate = $state(currentDate)
+	let showYearPicker = $state(false)
+	let showDatePicker = $state(false)
+	let size: 16 | 24 | 32 = $derived(dimension === 'large' ? 32 : dimension === 'small' ? 16 : 24)
+	let variant: Variant = $derived(
+		dimension === 'large' ? 'large' : dimension === 'small' ? 'small' : 'default',
+	)
+	let daysInMonth = $derived(new Date(selectedYear, selectedMonth + 1, 0).getDate())
+	let firstDayOfMonth = $derived((new Date(selectedYear, selectedMonth, 1).getDay() + 6) % 7)
+	let prevMonthDays = $derived(new Date(selectedYear, selectedMonth, 0).getDate())
+	let totalCells = 42
+	let calendarDays = $derived(
+		Array.from({ length: totalCells }, (_, i) => {
+			const dayNumber = i - firstDayOfMonth + 1
+			if (dayNumber <= 0) {
+				return {
+					date: prevMonthDays + dayNumber,
+					type: 'prev',
+				}
+			} else if (dayNumber > daysInMonth) {
+				return {
+					date: dayNumber - daysInMonth,
+					type: 'next',
+				}
+			} else {
+				return {
+					date: dayNumber,
+					type: 'current',
+				}
+			}
+		}),
+	)
+
+	let datePicker: HTMLDivElement | undefined
+	let selectedYearElement: HTMLSpanElement | undefined = $state(undefined)
+
+	const months = getLocalMonthNames()
+	const days = getLocalDayNames()
+
+	function getLocalMonthNames() {
+		const formatter = new Intl.DateTimeFormat(undefined, { month: 'long' })
+		const monthNames = []
+		for (let month = 0; month < 12; month++) {
+			const date = new Date(2020, month, 1)
+			monthNames.push(formatter.format(date))
+		}
+		return monthNames
+	}
+	function getLocalDayNames() {
+		const formatter = new Intl.DateTimeFormat(undefined, { weekday: 'short' })
+		const dayNames = []
+		for (let day = 0; day < 7; day++) {
+			const date = new Date(2020, 0, day - 1)
+			dayNames.push(formatter.format(date))
+		}
+		return dayNames
+	}
+
+	function changeMonth(direction: number) {
+		selectedMonth += direction
+		if (selectedMonth < 0) {
+			selectedMonth = 11
+			selectedYear--
+		} else if (selectedMonth > 11) {
+			selectedMonth = 0
+			selectedYear++
+		}
+	}
+
+	function changeYear(year: number) {
+		selectedYear = year
+		selectDate(currentDate.getDate())
+		showYearPicker = false
+	}
+
+	function selectDate(date: number) {
+		selectedDate = new Date(selectedYear, selectedMonth, date)
+	}
+
+	function isSelected(date: number) {
+		return (
+			selectedDate.getDate() === date &&
+			selectedDate.getMonth() === selectedMonth &&
+			selectedDate.getFullYear() === selectedYear
+		)
+	}
+
+	function isCurrentDay(date: number) {
+		return (
+			date === currentDate.getDate() &&
+			selectedMonth === currentDate.getMonth() &&
+			selectedYear === currentDate.getFullYear()
+		)
+	}
+
+	function formatDate(date: Date) {
+		const year = date.getFullYear()
+		const month = String(date.getMonth() + 1).padStart(2, '0')
+		const day = String(date.getDate()).padStart(2, '0')
+		return `${year}-${month}-${day}`
+	}
```

Comment:
I don't think we need this formatting. Probably we want to leave the default locale formatting but to be verified. @agazso @david-gauquelin any opinion?

Reply:
As far as I can see this is only used for the selected date value, not for displaying.

This is not needed, it should just use the `Date` value, any formatting should happen in the invoking user's code.